trin-core/src/utp/stream.rs

```diff
@@ -717,7 +711,7 @@ impl UtpSocket {
             // When SYN is send and we receive STATE, do not reply
             (SocketState::SynSent, PacketType::State) => {
                 self.connected_to = src;
-                self.ack_nr = packet.seq_nr();
+                self.ack_nr = packet.seq_nr() - 1;
```

Comment:
It may be better to address this deviation from the BEP29 spec in a separate, tiny PR as it seems unrelated to the rest. I don't feel too strongly about this though.

Reply:
Sending OFFER messages and init uTP stream to Fluffy breaks if we don't address it here :) I'll create an issue to not forget to add uTP tests that catch those deviations.